### PR TITLE
feat: persists chat window size on reload

### PR DESCRIPTION
**Pull request by ghgoodreau**
## Motivation

Issue #47. It would be nice to persist the chat size on revisits for consistency.

## Solution

Utilizing the handy useLocalStorage util, we can save and update the chat window size whenever it's changed. If the key doesn't exist in localStorage, we default to the original 50% width.

## Checklist

- [x] Tested in Chrome
- [x] Tested in Safari

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -1016,7 +1026,10 @@ function App() {
               bottomLeft: false,
               topLeft: false,
             }}
-            onResizeStop={autoZoomIfNecessary}
+            onResizeStop={(_, __, ref) => {
+              setSavedChatSize(ref.style.width);
+              autoZoomIfNecessary();
+            }}
```
Comment (transmissions11) [inline src/components/App.tsx:1032]: now that i think of it lets add a mixpanel event here

File: src/utils/constants.ts [hunk 2]
```diff
@@ -61,5 +61,7 @@ export const CODE_BLOCK_LANGUAGE_DETECT_REGEX = /^```[a-zA-Z0-9-]*$/m;
 export const STREAM_CANCELED_ERROR_MESSAGE = "STREAM_CANCELED";
 export const STALE_STREAM_ERROR_MESSAGE = "STALE_STREAM";
 
+export const SAVED_CHAT_SIZE_LOCAL_STORAGE_KEY = "FLUX_SAVED_CHAT_SIZE";
```
Comment (transmissions11) [inline src/utils/constants.ts:64]: can you put this up with the other "local_storage_key" vars

File: src/components/App.tsx [hunk 3]
```diff
@@ -923,6 +924,14 @@ function App() {
 
   useDebouncedWindowResize(autoZoomIfNecessary, 100);
 
+  /*//////////////////////////////////////////////////////////////
+                        CHAT RESIZE LOGIC
+  //////////////////////////////////////////////////////////////*/
+
+  const [savedChatSize, setSavedChatSize] = useLocalStorage<string>(
+    SAVED_CHAT_SIZE_LOCAL_STORAGE_KEY
+  );
```
Comment (transmissions11) [inline src/components/App.tsx:933]: wait will this be a string? is ref.style.width a value like `533px` or `533`

Comment (ghgoodreau) [reply src/components/App.tsx:933]: It's a string representing a percentage value, so this is correct

Screenshot 2023-04-04 at 4 15 20 PM
Screenshot 2023-04-04 at 4 14 35 PM

File: src/utils/constants.ts (hunk 2 above)
Comment (ghgoodreau) [reply src/utils/constants.ts:64]: yup updating now

File: src/components/App.tsx (hunk 1 above)
Comment (ghgoodreau) [reply src/components/App.tsx]: added!

File: src/components/App.tsx (hunk 3 above)
Comment (transmissions11) [reply src/components/App.tsx:933]: ahh interesting — sg

File: src/components/App.tsx [hunk 4]
```diff
@@ -1016,7 +1026,11 @@ function App() {
               bottomLeft: false,
               topLeft: false,
             }}
-            onResizeStop={autoZoomIfNecessary}
+            onResizeStop={(_, __, ref) => {
+              setSavedChatSize(ref.style.width);
+              autoZoomIfNecessary();
+              if (MIXPANEL_TOKEN) mixpanel.track("Resized chat window");
+            }}
```
Comment (transmissions11) [inline src/components/App.tsx:1033]:
```suggestion

              if (MIXPANEL_TOKEN) mixpanel.track("Resized chat window");
            }}
```

File: src/components/App.tsx [hunk 5]
```diff
@@ -1003,7 +1012,8 @@ function App() {
             maxWidth="75%"
             minWidth="15%"
             defaultSize={{
-              width: "50%",
+              // defaults to the previously used chat size if it exists.
```
Comment (transmissions11) [inline src/components/App.tsx:1015]:
```suggestion
              // Defaults to the previously used chat size if it exists.
```

**Comment by transmissions11** [comment]
lgtm — thanks so much

**Pull request merged**